lib/kubernetes-deploy/kubernetes_resource.rb

```diff
@@ -421,6 +422,10 @@ def global?
       @global || self.class::GLOBAL
     end
 
+    def prefetch_kinds_on_sync
```

Comment:
Nit: it isn't really necessary to expose this, since you necessarily can get the classes from the resources, i.e.:

```ruby
      dependencies = resources.map(&:class).uniq.flat_map do |c|
        c.const_get(:SYNC_DEPENDENCIES)
      end
      kinds = (resources.map(&:kubectl_resource_type) + dependencies).compact.uniq
```